### ISSUE TITLE
Debian Jessie has end of security support in May 2018

### ISSUE THREAD

**Issue by chauffer** [docker-library/python#264]
According to https://en.wikipedia.org/wiki/Debian_version_history#Release_table, Debian Jessie will lose security support May of this year.

Are you planning to upgrade these images? They're all based on Jessie.

```
simone•~» for i in python:2 python:3 python:2-slim python:3-slim; do docker run $i cat /etc/issue.net; done
Debian GNU/Linux 8
Debian GNU/Linux 8
Debian GNU/Linux 8
Debian GNU/Linux 8
```

**Comment by tianon** [comment]
:wink: (see https://github.com/docker-library/python/pull/208 and https://github.com/docker-library/python/pull/233)

```console
$ bashbrew list --uniq python | grep stretch
python:3.7.0b1-stretch
python:3.7.0b1-slim-stretch
python:3.6.4-stretch
python:3.6.4-slim-stretch
python:2.7.14-stretch
python:2.7.14-slim-stretch
```

Full list of all stretch-related aliases of the above unique tags:

```console
$ bashbrew list python | grep stretch
python:3.7.0b1-stretch
python:3.7-rc-stretch
python:rc-stretch
python:3.7.0b1-slim-stretch
python:3.7-rc-slim-stretch
python:rc-slim-stretch
python:3.6.4-stretch
python:3.6-stretch
python:3-stretch
python:stretch
python:3.6.4-slim-stretch
python:3.6-slim-stretch
python:3-slim-stretch
python:slim-stretch
python:2.7.14-stretch
python:2.7-stretch
python:2-stretch
python:2.7.14-slim-stretch
python:2.7-slim-stretch
python:2-slim-stretch
```

**Comment by chauffer** [comment]
Will the default be stretch?

**Comment by tianon** [comment]
At some point, yes; transition timeline TBD (especially since it's a bit of
a breaking change for a lot of folks).

**Comment by miksuh** [comment]
Actually that is not really true. All Debian stable releases are LTS (Long Time Support) releases these days. Debian LTS team gives atleast 5 years of support to every Debian stable release including Debian Jessie. When support from the Debian security team ends Debian LTS team will take over the control and they will continue to support the Debian release. E.g the Debian Jessie will be supported by Debian LTS team until atleast mid 2020. Debian 8.0 "Jessie" was released in  April 2015, so you can expect that Debian LTS team will support it until atleast April 2020. Similarly Debian 9.0 "Stretch" was released in summer 2017, so you can expect that Debian LTS team will support it until atleast summer 2022.

You can read about Debian LTS support here:

https://wiki.debian.org/LTS

**Comment by macropin** [comment]
@tianon it looks like the floating tag `python:2.7` now refers to `python:2.7-stretch` rather than `python:2.7-jessie`. Where can I track changes like this?

**Comment by tianon** [comment]
@macropin correct -- for every image update, there'll be a PR here (https://github.com/docker-library/python), followed by a PR to `library/python` in https://github.com/docker-library/official-images (https://github.com/docker-library/official-images/pulls?q=label%3Alibrary%2Fpython).

See also "Image updates:" under "Quick reference" on https://hub.docker.com/_/python/.

**Comment by w-flo** [comment]
I have a related question, it seems like this commit from 7 days ago changed the 3-slim tag to point to the stretch-based version instead of the jessie-based one (even though it's not mentioned in the commit message): https://github.com/docker-library/official-images/commit/f200d7ffdeb698ad62802351dd9e023cad57a4bd#diff-051bb6df61b6c6e7f8c1868985011b07

But it's not in the docker hub registry yet. Is there some delay for review? If so, is there any place where I can track the process until the official-images repo is in sync with the images published on hub.docker.org? (At least some of) the other changes in this commit are already live, e.g. "alpine" points to the alpine-3.7 based version instead of the previous alpine-3.4. Or maybe that's a bug in the git repo -> docker hub sync tool and the new 3-slim tag should already be live?

**Comment by yosifkit** [comment]
Thanks @w-flo, it looks like our logic to avoid no-op pushes to the Docker Hub is causing tag swapping to not work. :thinking: We'll try to get it fixed as soon as possible.

**Comment by chauffer** [comment]
Thanks @yosifkit!

**Comment by tianon** [comment]
A proper fix for that issue is going to be a little complicated, so
I've gone and manually pushed the offending tags now.  Should be
fixed!

**Comment by chauffer** [comment]
Thanks @tianon!

```simone•~» for i in python:2 python:3 python:2-slim python:3-slim; do docker run $i cat /etc/issue.net; done
Debian GNU/Linux 9
Debian GNU/Linux 9
Debian GNU/Linux 9
Debian GNU/Linux 9```

**Comment by tbreeds** [comment]
I don't know if this should be a separate issue.

Can we get stretch 3.5 images?

I've cloned this repo, copied 3.5/jessie -> 3.5/stretch and up update.sh
building the 3.5/stretch/Dockerfile works.

Would a PR to do that be appropriate?

**Comment by yosifkit** [comment]
@tbreeds, there is a PR open: #308. :wink:

**Comment by tbreeds** [comment]
@yosifkit Oh thanks!

**Comment by tianon** [comment]
Fixed by #308 :+1:
